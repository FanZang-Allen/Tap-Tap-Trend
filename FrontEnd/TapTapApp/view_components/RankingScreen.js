import React from 'react';
import {useState, useEffect} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {Picker} from '@react-native-picker/picker';
import {baseUrl} from '../model/Config';
import GameList from '../model/GameList';
import GameFlatList from './GameFlatList';

const RankingScreen = ({navigation, testMode}) => {
  const [selectedRegion, setSelectedRegion] = useState('USA');
  const [gameData, setgameData] = useState(null);
  const regionUrl = baseUrl + '/api/rank?region=';

  async function fetchData() {
    const gameClass = new GameList(regionUrl + selectedRegion, null);
    await gameClass.FetchData();
    if (gameClass.success === false) {
      setgameData(undefined);
    } else {
      setgameData(gameClass.GetData());
    }
  }

  async function switchRegion(itemValue) {
    setSelectedRegion(itemValue);
    setgameData(null);
    fetchData();
  }

  useEffect(() => {
    let fetchCalled = false;
    async function intialFetch() {
      const gameClass = new GameList(regionUrl + selectedRegion, null);
      await gameClass.FetchData();
      if (gameClass.success === false) {
        setgameData(undefined);
      } else {
        setgameData(gameClass.GetData());
      }
    }
    if (fetchCalled === false) {
      intialFetch();
      fetchCalled = true;
    }
  }, [regionUrl, selectedRegion]);

  return (
    <View style={styles.container}>
      <View style={styles.header_container}>
        <Text style={styles.header}>Top 15 Mobile Games</Text>
        <Picker
          selectedValue={selectedRegion}
          onValueChange={switchRegion}
          style={styles.region_picker}>
          <Picker.Item label="USA" value="USA" />
          <Picker.Item label="World" value="World" />
          <Picker.Item label="China" value="Taiwan" />
          <Picker.Item label="Japan" value="Japan" />
          <Picker.Item label="Korea" value="Korea" />
          <Picker.Item label="India" value="India" />
          <Picker.Item label="Vietnam" value="Vietnam" />
        </Picker>
      </View>
      <GameFlatList
        gotData={gameData}
        navigation={navigation}
        refreshing={false}
        onRefresh={() => console.log('This screen not suppport refresh')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header_container: {
    height: '10%',
    margin: 20,
  },
  header: {
    fontSize: 25,
    fontWeight: 'bold',
    alignSelf: 'center',
    color: '#000000',
  },
  region_picker: {
    width: '45%',
    alignSelf: 'center',
  },
  background: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 20,
    paddingBottom: 10,
  },
});

export default RankingScreen;
