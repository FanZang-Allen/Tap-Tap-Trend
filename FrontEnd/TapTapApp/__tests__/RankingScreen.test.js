import 'react-native';
import React from 'react';
import RankingScreen from '../view_components/RankingScreen';
import renderer from 'react-test-renderer';

test('Ranking Rendered Page', () => {
  const snap = renderer.create(<RankingScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
