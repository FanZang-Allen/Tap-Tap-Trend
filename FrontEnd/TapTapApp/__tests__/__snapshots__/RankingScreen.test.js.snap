// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ranking Rendered Page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "10%",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#000000",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Top 15 Mobile Games
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": "45%",
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "USA",
              "testID": undefined,
              "textColor": undefined,
              "value": "USA",
            },
            Object {
              "label": "World",
              "testID": undefined,
              "textColor": undefined,
              "value": "World",
            },
            Object {
              "label": "China",
              "testID": undefined,
              "textColor": undefined,
              "value": "Taiwan",
            },
            Object {
              "label": "Japan",
              "testID": undefined,
              "textColor": undefined,
              "value": "Japan",
            },
            Object {
              "label": "Korea",
              "testID": undefined,
              "textColor": undefined,
              "value": "Korea",
            },
            Object {
              "label": "India",
              "testID": undefined,
              "textColor": undefined,
              "value": "India",
            },
            Object {
              "label": "Vietnam",
              "testID": undefined,
              "textColor": undefined,
              "value": "Vietnam",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9B9B9B",
          "flex": 4,
          "fontSize": 14,
        }
      }
    >
      Loading
    </Text>
  </View>
</View>
`;
